azure_monitor/azfunc_sample/README.md

```diff
@@ -0,0 +1,86 @@
+---
+page_type: sample
+languages:
+- python
+products:
+- azure
+description: "This sample contains a simple Azure Function to show how you can send correlated application log information as well as information of external dependencies to Azure Monitor (Application Insights)."
+urlFragment: azure-monitor-opencensus-python
+---
+
+# Correlated End-To-End Azure Function (Python) Instrumentation with Opencensus 
+
+## Overview
+
+Azure Function (Python) already supports instrumentaton using Opencensus. Unfortunately, the current out-of-the-box (OOTB) implementation/support has some flaws (see below), which make it difficult to get a correlated end-to-end view of a single function invocation (in Azure Monitor/Application Insights).
+
+It is also important to assign the correct cloud role name and cloud instance name to different log records sent to Application Insights (for additional insights). This is especially relevant for distributed, scalable applications sending logs to a single Application Insights instance.
+
+### Example of a Correlated End-to-End Azure Function Invocation
+
+End-to-end transaction:
+
+![AI Correlated end2end function invocation](./docs/assets/01-ai-correlated-invocation.PNG)
+
+Associated application trace messages:
+
+![AI Application trace messages](./docs/assets/02-ai-correlated-invocation.PNG)
+
+Associated AI Application Map:
+
+![AI Application Map](./docs/assets/03-ai-application-map.PNG)
+
+The images shows the following elements:
+
+- Correlated (per Azure Function invocatio) log information incl
+    - Incoming request information
+    - Appication traces & error messages
+    - External dependencies (IN-PROC & HTTP using Python's requests)
+- Application Map (defined cloud role and cloud instance values)
+
+### Known Limitations in Current OOTB Logging Integration (Azure Function)
+
+- External dependency log records don't have the correct *Cloud Role Name* (field: cloud_RoleName). This results in wrong information displayed in Application Insight's *Application Map*. See <https://github.com/census-instrumentation/opencensus-python/issues/1052>
+- Errors (logged with the OOTB logging integration) are not correctly logged as Application Insights Exception records. Currently `logging.exception(..)` creates a trace record (with severity level 3 = Error). This comes with certain side effects: E.g. errors are won't show up in Application Insights dashboard (*Failures*) and troubleshooting becomes harder (see issue <https://github.com/Azure/azure-functions-python-worker/issues/866>)
```

Comment:
Seems like `OOTB logging integration` in this section is referring to the Azure functions worker logger, not the Opencensus logger. The `exceptions` limitation is an issue in the functions logger, but the OpenCensus logger does not have this issue. With that being said, users will be getting duplicate telemetry if they start using the OpenCensus logger as well (since both loggers send user generated logs). We are developing a feature to disable this double telemetry.

Reply:
thanks for the update. I noticed the "duplicated log entry" issue. therefore, I explicitly disable log propagation in the sample code. but good to know that there will be a feature in future.